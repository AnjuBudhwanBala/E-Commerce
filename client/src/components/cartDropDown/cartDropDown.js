import React, { useCallback } from 'react';

import CartItem from '../cartItem/cartItem';
import { useSelector, shallowEqual, useDispatch } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { selectCartItems } from '../../redux/cart/cartSelector';
import { toggleCartHidden } from '../../redux/cart/cart.actions';
import {
  CartDropDownContainer,
  EmptyMessageContainer,
  CartItemsContainer,
  CartDropdownButton
} from './cartDropDown.styles';

const CartDropDown = ({ history, ...otherProps }) => {
  const itemsInCart = useSelector(selectCartItems, shallowEqual);
  const dispatch = useDispatch();
  const cartHidden = useCallback(() => dispatch(toggleCartHidden()), [
    dispatch
  ]);

  return (
    <CartDropDownContainer>
      <CartItemsContainer>
        {itemsInCart.length ? (
          itemsInCart.map(cartItem => (
            <CartItem key={cartItem.id} item={cartItem} />
          ))
        ) : (
          <EmptyMessageContainer>Your cart is empty</EmptyMessageContainer>
        )}
      </CartItemsContainer>
      <CartDropdownButton
        onClick={() => {
          history.push('/checkout');
          cartHidden();
        }}
      >
        GO TO CHECKOUT
      </CartDropdownButton>
    </CartDropDownContainer>
  );
};

export default withRouter(CartDropDown);
