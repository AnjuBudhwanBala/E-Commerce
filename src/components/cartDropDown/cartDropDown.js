import React, { useCallback } from 'react';
import './cartDropDown.styles.scss';
import CustomButton from '../customButton/customButton';
import CartItem from '../cartItem/cartItem';
import { useSelector, shallowEqual, useDispatch } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { selectCartItems } from '../../redux/cart/cartSelector';
import { toggleCartHidden } from '../../redux/cart/cart.actions';

const CartDropDown = ({ history, ...otherProps }) => {
  const itemsInCart = useSelector(selectCartItems, shallowEqual);
  const dispatch = useDispatch();
  const cartHidden = useCallback(() => dispatch(toggleCartHidden()), [
    dispatch
  ]);

  return (
    <div className="cart-dropdown">
      <div className="cart-items">
        {itemsInCart.length ? (
          itemsInCart.map(cartItem => (
            <CartItem key={cartItem.id} item={cartItem} />
          ))
        ) : (
          <span className="empty-message">Your cart is empty</span>
        )}
      </div>
      <CustomButton
        onClick={() => {
          history.push('/checkout');
          cartHidden();
        }}
      >
        GO TO CHECKOUT
      </CustomButton>
    </div>
  );
};

export default withRouter(CartDropDown);
