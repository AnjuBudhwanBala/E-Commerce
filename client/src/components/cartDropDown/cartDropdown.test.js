import React from 'react';
import { shallow } from 'enzyme';
import { CartDropdown } from './cartDropDown';

describe('CartDropDown component', () => {
  let wrapper;
  let mockHistory;
  let mockDispatch;
  const mockCartItems = [{ id: 1 }, { id: 2 }, { id: 3 }];

  beforeEach(() => {
    mockHistory = {
      push: jest.fn()
    };

    mockDispatch = jest.fn();

    const mockProps = {
      history: mockHistory,
      dispatch: mockDispatch,
      cartItems: mockCartItems
    };

    wrapper = shallow(<CartDropdown {...mockProps} />);
  });

  it('should render CartDropdown component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
