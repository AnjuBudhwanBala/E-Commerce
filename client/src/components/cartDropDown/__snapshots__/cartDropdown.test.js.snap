// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartDropDown component should render CartDropdown component 1`] = `ShallowWrapper {}`;
